### PR TITLE
HPCC-19479 Create WsELK for ELK Stack integration

### PR DESCRIPTION
**Pull request by rpastrana**
- Declares ECLWatch configuration for ELK Stack integration
- Provides integration configuration

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by rpastrana** [comment]
@miguelvazq please confirm this service provides the information you need to embed Kibana dashboards
You can expect the information to be returned in this format:
{
  "GetConfigDetailsResponse": {
    "IntegrateKibana": true,
    "KibanaAddress": "String",
    "KibanaPort": "String",
    "KibanaEntryPointURI": "String",
    "ReportElasticSearchHealth": true,
    "ElasticSearchAdresses": "String",
    "ElasticSearchPort": "String",
    "ReportLogStashHealth": true,
    "LogStashAddress": "String",
    "LogStashPort": "String"
  }
}

**Comment by miguelvazq** [comment]
This could grow as we keep think of coming up with feature sets. But, at first glace and our discussion looks good. Only concern would be a security issue sending requests via ajax to IP's that are locked down.

**Comment by rpastrana** [comment]
@wangkx please review.

File: esp/scm/ws_elk.ecm [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    string ElasticSearchAdresses;
+    string ElasticSearchPort;
+    boolean ReportLogStashHealth;
+    string LogStashAddress;
+    string LogStashPort;
+ };
+
+ESPservice [auth_feature("ESDLConfigAccess:ACCESS"), version("1.0"), default_client_version("1.0"), exceptions_inline("xslt/exceptions.xslt")] ws_elk
+{
+    ESPmethod getConfigDetails(GetConfigDetailsRequest, GetConfigDetailsResponse);
```
Comment (wangkx) [inline esp/scm/ws_elk.ecm]: I prefer that the method name starts with Capital ('G') as other ESP services.

File: initfiles/componentfiles/configxml/espsmcservice.xsd.in [hunk 2]
```diff
@@ -49,10 +49,10 @@
...
             </xs:attribute>
                 <xs:annotation>
-	          <xs:appinfo>
-	          <!--DOCS_Autobuild_Code-->
-          	    <docid>SMC-T01</docid>
-	          </xs:appinfo>
+              <xs:appinfo>
+              <!--DOCS_Autobuild_Code-->
+                <docid>SMC-T01</docid>
+              </xs:appinfo>
```
Comment (wangkx) [inline initfiles/componentfiles/configxml/espsmcservice.xsd.in]: wrong indent in at least 1 line here

File: initfiles/componentfiles/configxml/espsmcservice.xsd.in [hunk 3]
```diff
@@ -153,15 +153,99 @@
...
+                <xs:appinfo>
+                    <tooltip>Declares the target LogStash nodes' port</tooltip>
+                </xs:appinfo>
+            </xs:annotation>
+        </xs:attribute>
+    </xs:attributeGroup>
     <xs:attributeGroup name="Monitoring">
          <xs:annotation>
-	    <xs:appinfo>
+        <xs:appinfo>
```
Comment (wangkx) [inline initfiles/componentfiles/configxml/espsmcservice.xsd.in]: wrong indent

**Review by wangkx — COMMENTED** [review]
@rpastrana  just a few minor comments.

File: esp/scm/ws_elk.ecm (hunk 1 above)
Comment (rpastrana) [reply esp/scm/ws_elk.ecm]: Indeed! Not sure how that slipped through

File: initfiles/componentfiles/configxml/espsmcservice.xsd.in [hunk 4]
```diff
@@ -49,10 +49,10 @@
                 </xs:annotation>
             </xs:attribute>
                 <xs:annotation>
-	          <xs:appinfo>
-	          <!--DOCS_Autobuild_Code-->
-          	    <docid>SMC-T01</docid>
-	          </xs:appinfo>
+              <xs:appinfo>
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/espsmcservice.xsd.in]: @g-pan @JamesDeFabia please comment on what this element should be and if there are any missing docid elements, etc.

**Comment by rpastrana** [comment]
@wangkx please review.

**Review by wangkx — APPROVED** [review]
@rpastrana looks good.

**Pull request merged**